### ISSUE TITLE
Top View Map Appeareance options not defined with web map services

### ISSUE THREAD

**Issue by ReimarBauer** [Open-MSS/MSS#401]
**[Original report](https://bitbucket.org/wxmetvis/mss/issue/401) by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

The options form of TopView has elements which can't be used when images from a web map services used.

* [ ] fill map background  Color
* [ ] fill continents Color

We have to disable these elements for this usecase.

**Comment by ReimarBauer** [comment]
**Original comment by Shivashis Padhi (Bitbucket: [plant99](https://bitbucket.org/plant99), GitHub: [plant99](https://github.com/plant99)).**

----------------------------------------

Working on this.

**Comment by ReimarBauer** [comment]
**Original comment by Shivashis Padhi (Bitbucket: [plant99](https://bitbucket.org/plant99), GitHub: [plant99](https://github.com/plant99)).**

----------------------------------------

![Screenshot from 2019-04-25 20-12-46.png](https://bitbucket.org/repo/MBp6Xa/images/1231041230-Screenshot%20from%202019-04-25%2020-12-46.png)

@ReimarBauer  Can you check if "fill map background" is setChecked(True) and disabled in develop branch in your local too?

**Comment by ReimarBauer** [comment]
**Original comment by Shivashis Padhi (Bitbucket: [plant99](https://bitbucket.org/plant99), GitHub: [plant99](https://github.com/plant99)).**

----------------------------------------

https://bitbucket.org/wxmetvis/mss/src/595aec9ed7ec2e89eadf86648cc2bf16827fd079/mslib/msui/qt5/ui_topview_mapappearance.py#lines-32

Basically I have to take care of just "fill continents" right?

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

Only the color for the waypoints is handled by the gui.![Bildschirmfoto von »2019-04-26 08-13-20«.png](https://bitbucket.org/repo/MBp6Xa/images/1244450032-Bildschirmfoto%20von%20%C2%BB2019-04-26%2008-13-20%C2%AB.png)

fill map background is disabled, but the color box chooser right of it not.

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

thought I had seen continents filled, but not in that example.

**Comment by ReimarBauer** [comment]
**Original comment by Shivashis Padhi (Bitbucket: [plant99](https://bitbucket.org/plant99), GitHub: [plant99](https://github.com/plant99)).**

----------------------------------------

So an ideal functionality would be, to 
1. uncheck 'fill map background' and disable 'fill continents' when wms connected
2. check 'fill map background', and enable 'fill continents' when wms disconnected.

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

yes, that should be a small change.

If possible try to find the changesets which had added this feature. 
It looks like half implemented, if there is no option to call those images with different colors from the server.  May be it was and now the problem is on another part of the codebase.

Then it depends on the amount of changes to get that working again for fixing in stable.

**Comment by ReimarBauer** [comment]
**Original comment by Shivashis Padhi (Bitbucket: [plant99](https://bitbucket.org/plant99), GitHub: [plant99](https://github.com/plant99)).**

----------------------------------------

Sure will explore more before making a PR.

**Comment by ReimarBauer** [comment]
**Original comment by Shivashis Padhi (Bitbucket: [plant99](https://bitbucket.org/plant99), GitHub: [plant99](https://github.com/plant99)).**

----------------------------------------

```
#!python

grep -nr "cbFillWaterBodies" .
Binary file ./mslib/msui/qt5/__pycache__/ui_topview_mapappearance.cpython-37.pyc matches
./mslib/msui/qt5/ui_topview_mapappearance.py:31:        self.cbFillWaterBodies = QtWidgets.QCheckBox(MapAppearanceDialog)
./mslib/msui/qt5/ui_topview_mapappearance.py:32:        self.cbFillWaterBodies.setEnabled(False)
./mslib/msui/qt5/ui_topview_mapappearance.py:33:        self.cbFillWaterBodies.setMinimumSize(QtCore.QSize(145, 0))
./mslib/msui/qt5/ui_topview_mapappearance.py:62:        self.cbFillWaterBodies.setPalette(palette)
./mslib/msui/qt5/ui_topview_mapappearance.py:63:        self.cbFillWaterBodies.setChecked(True)
./mslib/msui/qt5/ui_topview_mapappearance.py:64:        self.cbFillWaterBodies.setObjectName("cbFillWaterBodies")
./mslib/msui/qt5/ui_topview_mapappearance.py:65:        self.horizontalLayout.addWidget(self.cbFillWaterBodies)
./mslib/msui/qt5/ui_topview_mapappearance.py:160:        self.cbFillWaterBodies.setText(_translate("MapAppearanceDialog", "fill map background"))
Binary file ./mslib/msui/__pycache__/topview.cpython-37.pyc matches
./mslib/msui/topview.py:81:        self.cbFillWaterBodies.setChecked(settings_dict["fill_waterbodies"])
./mslib/msui/topview.py:108:            "fill_waterbodies": self.cbFillWaterBodies.isChecked(),
./mslib/msui/ui/ui_topview_mapappearance.ui:43:      <widget class="QCheckBox" name="cbFillWaterBodies">

```


```
#!python

grep -nr "fill map background" .
Binary file ./mslib/msui/qt5/__pycache__/ui_topview_mapappearance.cpython-37.pyc matches
./mslib/msui/qt5/ui_topview_mapappearance.py:160:        self.cbFillWaterBodies.setText(_translate("MapAppearanceDialog", "fill map background"))
./mslib/msui/ui/ui_topview_mapappearance.ui:145:        <string>fill map background</string>

```

I believe this has no dependent code snippet. It might be half-implemented. I will PR the simple update.

**Comment by ReimarBauer** [comment]
**Original comment by Shivashis Padhi (Bitbucket: [plant99](https://bitbucket.org/plant99), GitHub: [plant99](https://github.com/plant99)).**

----------------------------------------

Also, this change(enabling and disabling of check-boxes), should they happen after 'get_capabilities' or 'first getmap'

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

we do it as early as possible, so after 'get_capabilities' is the better place.

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

The options form of TopView has elements which can't be used when images from a web map services used.

* [ ] fill map background  Color
* [ ] fill continents Color

We have to disable these elements for this usecase.

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

Merged in plant99/mss/i401 (pull request #636)

Fixes#401

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

The options form of TopView has elements which can't be used when images from a web map services used.

* [ ] fill map background  Color
* [ ] fill continents Color

We have to disable these elements for this usecase.
